File: API/HanoConnect.API/HanoConnect.API/DTOs/OpportunityResponseDto.cs
using HanoConnect.API.Models;
using System;
using System.Collections.Generic;

namespace HanoConnect.API.DTOs
{
    public class OpportunityResponseDto
    {
        public int OpportunityId { get; set; }
        public string Title { get; set; } = string.Empty;
        public string? Description { get; set; }
        public string? Location { get; set; }
        public DateTime? StartDate { get; set; }
        public DateTime? EndDate { get; set; }
        public bool IsFlexibleTime { get; set; }
        public int? RequiredVolunteers { get; set; }
        public string? Benefits { get; set; }
        public string? ContactInfo { get; set; }
        public DateTime? ApplicationDeadline { get; set; }
        public string Status { get; set; } = "Open";
        public bool IsApprovedByAdmin { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }

        // Thông tin liên quan từ Organization
        public int OrganizationId { get; set; }
        public string OrganizationName { get; set; } = string.Empty;
        public string? OrganizationContactPerson { get; set; }

        // Thông tin liên quan từ Cause
        public int CauseId { get; set; }
        public string CauseName { get; set; } = string.Empty;

        // Danh sách kỹ năng
        public List<SkillDto> Skills { get; set; } = new List<SkillDto>();
    }

    public class SkillDto
    {
        public int SkillId { get; set; }
        public string SkillName { get; set; } = string.Empty;
    }
}

File: HanoConnectAdmin/app/src/main/java/com/example/hanoconnectadmin/models/OpportunityResponseDto.java
package com.example.hanoconnectadmin.models;

import com.google.gson.annotations.SerializedName;
import java.util.List;

public class OpportunityResponseDto {

    @SerializedName("opportunityId")
    private int opportunityId;

    @SerializedName("title")
    private String title;

    @SerializedName("description")
    private String description;

    @SerializedName("location")
    private String location;

    @SerializedName("startDate")
    private String startDate;

    @SerializedName("endDate")
    private String endDate;

    @SerializedName("isFlexibleTime")
    private boolean isFlexibleTime;

    @SerializedName("requiredVolunteers")
    private Integer requiredVolunteers; // Dùng Integer để chấp nhận giá trị null

    @SerializedName("benefits")
    private String benefits;

    @SerializedName("contactInfo")
    private String contactInfo;

    @SerializedName("applicationDeadline")
    private String applicationDeadline;

    @SerializedName("status")
    private String status;

    @SerializedName("isApprovedByAdmin")
    private boolean isApprovedByAdmin;

    @SerializedName("createdAt")
    private String createdAt;

    @SerializedName("updatedAt")
    private String updatedAt;

    @SerializedName("organizationId")
    private int organizationId;

    @SerializedName("organizationName")
    private String organizationName;

    @SerializedName("organizationContactPerson")
    private String organizationContactPerson;

    @SerializedName("causeId")
    private int causeId;

    @SerializedName("causeName")
    private String causeName;

    @SerializedName("skills")
    private List<SkillDto> skills;

    // Getters cho các trường quan trọng để hiển thị
    public int getOpportunityId() { return opportunityId; }
    public String getTitle() { return title; }
    public String getOrganizationName() { return organizationName; }
    public String getLocation() { return location; }
}